src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -798,7 +961,21 @@ object LazyList extends SeqFactory[LazyList] {
   private def stateFromColl[A](coll: IterableOnce[A]): State[A] =
     stateFromCollConcatSuffix(coll)(State.Empty)
 
-  def empty[A]: LazyList[A] = _empty
+  override def concat[A](xss: collection.Iterable[A]*): LazyList[A] = {
+    if (xss.knownSize == 0) empty
+    else xss match {
+      case seq: collection.LinearSeq[collection.Iterable[A]] => newLL(concatLinearSeq(seq))
+      case _                                                 => newLL(concatIterator(xss.iterator))
+    }
+  }
+
+  private def concatLinearSeq[A](seq: collection.LinearSeq[collection.Iterable[A]]): State[A] =
+    if (seq.isEmpty) State.Empty
+    else stateFromCollConcatSuffix(seq.head)(concatLinearSeq(seq.tail))
+
+  private def concatIterator[A](it: Iterator[collection.Iterable[A]]): State[A] =
+    if (!it.hasNext) State.Empty
+    else stateFromCollConcatSuffix(it.next())(concatIterator(it))
 
   private final class WithFilter[A] private[LazyList](lazyList: LazyList[A], p: A => Boolean)
```

Comment:
I realized that `LazyList` fulfills the interface of `collection.WithFilter`, and is already lazy. Any objections to having `LazyList` extend `collection.WithFilter`, and then just having `withFilter` be an alias of `filter`?

Reply:
nvm, this doesn't work. it's a class